src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -671,30 +687,17 @@ Handle SystemDictionaryShared::get_shared_jar_manifest(int shared_path_index, TR
   Handle manifest ;
   if (shared_jar_manifest(shared_path_index) == NULL) {
     SharedClassPathEntry* ent = FileMapInfo::shared_path(shared_path_index);
-    long size = ent->manifest_size();
+    size_t size = (size_t)ent->manifest_size();
```

Comment:
size_t is unsigned, so I think you should change the following test to `if (size == 0)`